### PR TITLE
 [publication] fix notification

### PR DESCRIPTION
**Pull request by kongtiaowang**
## Brief summary of changes
add notification funtion for upload , review and submission.
#### Testing instructions (if applicable)
 upload , review or submit a proposal, then user who has the notification services will get the email.
#### Link(s) to related issue(s)
https://github.com/aces/Loris/issues/7204

### CODE REVIEW

**Comment by CamilleBeau** [comment]
Failed test suite. Looks like it can all be fixed with phpcbf

**Comment by CamilleBeau** [comment]
I tested this PR and it fixed the notification. I set permissions from publication to be received in my preferences, then I made a new account and uploaded a proposal & edited the proposal. I got notifications in both of these cases to my original account's email, but the notification implied that it was _my_ project proposal when I think it should refer to _a_ project proposal. 

Here are the messages received to the user that had notifications set. These messages seem like they are intended for the user who created the proposal.

> Hello,
> This is to confirm your project proposal "Testing a project" received on 2020-12-17.
> We will notify you once your proposal has been reviewed. In the mean time, you
> can edit your project proposal through the following link: /publication/view_project/?id=3.
> 
> Thank you,
> The ccna Team
> 

> Hello,
> Your project proposal "Testing a project" has been reviewed.
> You can view the details by clicking the following link /publication/view_project/?id=3.
> 
> Thank you,
> The ccna Team

**Comment by kongtiaowang** [comment]
@CamilleBeau Good catch! I have fixed the template file. It should be used to all the registed users not only the owner. Could u review again?

**Comment by CamilleBeau** [comment]
> @CamilleBeau Good catch! I have fixed the template file. It should be used to all the registed users not only the owner. Could u review again?

Looks good! Just need to fix for test suite again

File: smarty/templates/email/notifier_publication_review.tpl [hunk 1]
```diff
@@ -1,9 +1,9 @@
-Subject: [LORIS] Your project proposal has been reviewed
+Subject: [LORIS] A project proposal has been reviewed
 
 Hello,
 
-Your project proposal "{$Title}" has been reviewed.
+This project proposal "{$Title}" has been reviewed.
```
Comment (driusan) [inline smarty/templates/email/notifier_publication_review.tpl:5]: This sentence is grammatically incorrect. It should be "The project proposal {$Title} has been reviewed." not "This"

File: smarty/templates/email/notifier_publication_edit.tpl [hunk 2]
```diff
@@ -1,9 +1,9 @@
-Subject: [LORIS] Your project proposal has been edited
+Subject: [LORIS] A project proposal has been edited
 
 Hello,
 
-Your project proposal "{$Title}" has been edited by {$User}.
+This project proposal "{$Title}" has been edited by {$User}.
```
Comment (driusan) [inline smarty/templates/email/notifier_publication_edit.tpl:5]: This is also grammatically incorrect.

File: modules/publication/ajax/FileUpload.php [hunk 3]
```diff
@@ -446,26 +446,37 @@ function notify($pubID, $type) : void
...
-        Email::send(
-            $sendTo,
-            "notifier_publication_$type.tpl",
-            $emailData
+        $Notifier     = new \NDB_Notifier(
+            "publication",
+            $type
         );
-    }
+            $msg_data = [
```
Comment (driusan) [inline modules/publication/ajax/FileUpload.php:453]: This block looks like it's indented wrong.. it randomly goes up an indent level with no change in scope. I'm not sure why phpcs isn't flagging it.

**Review by driusan — CHANGES_REQUESTED** [review]

**Review by CamilleBeau — APPROVED** [review]

**Comment by ridz1208** [comment]
@CamilleBeau can you please put the pass manual test flag if you have tested the PR ?

**Review by ridz1208 — APPROVED** [review]

**Comment by ridz1208** [comment]
@driusan in your hands

**Pull request merged**